apps/ejabberd/src/node_push.erl

```diff
@@ -207,20 +208,16 @@ is_allowed_to_publish(PublishModel, Affiliation) ->
 parse_form(undefined) ->
     #{};
 parse_form(Form) ->
-    IsForm = ?NS_XDATA == exml_query:attr(Form, <<"xmlns">>),
-    IsSubmit = <<"submit">> == exml_query:attr(Form, <<"type">>),
```

Comment:
`type` is mandatory here. XEP-0357 specifies that this push node shall get `publish-options` form. The XEP-0060 (Publish-Subscribe) says that form inside of `publish-options` has to be the one defined in XEP-0004 (Data Forms). Finally XEP-0004 defines that `type` field of `x` element is `required` and has to be set to one of the following: `submit`, `cancel`, `result`, `form`. For reference see: [https://xmpp.org/extensions/xep-0004.html#schema](https://xmpp.org/extensions/xep-0004.html#schema)